core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala

```diff
@@ -154,7 +154,7 @@ object StreamletDeployment {
   val ServerAttributeName   = "server"
   val EndpointContainerPort = 3000
 
-  def name(appId: String, streamlet: String) = s"${appId}.${streamlet}"
+  def name(appId: String, streamlet: String) = s"${appId}.${streamlet}".take(63)
```

Comment:
My concern is just taking 63 characters may result in some name with a leading `-` or `_`. Is that allowed as a topic name in Kafka or for that matter a valid name in kubernetes ?

Reply:
Was the problem the topic name or the metadata label on the kafkatopic resource? I think this might not be the correct fix.

Reply:
It would be worse if the truncation results in two different streamlets getting the same topic.

Reply:
AFAIK the max length allowed for a Kafka topic is 249 characters

Reply:
> The name segment is required and must be 63 characters or less

Indeed it's the label name that exceeds this limit .. https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/